os_net_config/impl_ifcfg.py

```diff
@@ -340,13 +342,13 @@ def iproute2_apply_commands(self, dev_name, filename, data):
             if changes['DOMAIN'] == 'added':
                 commands.append(f"echo 'search {new}' >> /etc/resolv.conf")
             elif changes['DOMAIN'] == 'modified':
-                commands.append(f"sed -i -e 's/{old}/{new}/g' "
-                                "/etc/resolv.conf")
+                commands.append(f"sed -i -e 's/search[ ]*{old}/search {new}/g'"
+                                " /etc/resolv.conf")
             elif changes['DOMAIN'] == 'removed':
-                commands.insert("sed -i '/search/d' /etc/resolv.conf")
-        if 'DNS1' in changes:
-            # Remove nameservers from /etc/resolv.conf
-            commands.append("sed -i '/nameserver/d' /etc/resolv.conf")
+                commands.insert("sed -i '/^search/d' /etc/resolv.conf")
```

Comment:
This was modified. I need to check if either DNS1 or DNS2 is modified, and if so I need to remove both and add both new values. This was because I ran into issues if an IP moved from DNS1 to DNS2 or the reverse.